src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -670,8 +671,9 @@ def setter(hub):
         return LongRunningOperation(cmd.cli_ctx)(poller)
 
     if bool(identity_role) ^ bool(identity_scopes):
-        raise RequiredArgumentMissingError('At least one scope (--scopes) and one role (--role) required for system-managed identity role assignment')
-
+        raise RequiredArgumentMissingError('At least one scope (--scopes) and one role (--role) required for system-managed identity role assignment.')
+    if not system_identity and not user_identities:
+        raise RequiredArgumentMissingError('No identities provided to assign. Please provide system (--system) or user-assigned identities (--user-identities).')
```

Comment:
> or user-assigned identities (--user-identities).

Shouldn't the name of parameter `user_identities` of command `iot hub identity assign` be `'--user-assigned', '--user'` instead of `--user-identities`?

```
with self.argument_context('iot hub identity assign') as c:
        c.argument('system_identity', options_list=['--system-assigned', '--system'],
                   arg_type=get_three_state_flag(),
                   nargs='*', help="Assign a system-assigned managed identity to this hub.")
        c.argument('user_identities', options_list=['--user-assigned', '--user'],
                   nargs='*', help="Assign user-assigned managed identities to this hub. "
                   "Accept space-separated list of identity resource IDs.")
```